kernel/device.c

```diff
@@ -40,13 +40,13 @@ const struct device *z_impl_device_get_binding(const char *name)
 	 * performed. Reserve string comparisons for a fallback.
 	 */
 	STRUCT_SECTION_FOREACH(device, dev) {
-		if (z_device_is_ready(dev) && (dev->name == name)) {
+		if (z_impl_device_is_ready(dev) && (dev->name == name)) {
```

Comment:
why are you calling the implementation directly here and not using `device_is_ready` instead?

Reply:
yep

Reply:
do we agree that calling implementation (z_impl) in general should not be the norm and that any direct usage of the implementation function need some justification? If yes, this justifcation should be in the commit message, i.e. what @dcpleung  wrote above.

Reply:
@nashif I have update the commit msg. Please take a look again.